docs/langref.rst

```diff
@@ -415,18 +415,30 @@ Indirect function calls use a signature declared in the preamble.
 .. autoinst:: call_indirect
 .. autoinst:: func_addr
 
+.. _memory:
 
 Memory
 ======
 
 Cretonne provides fully general :inst:`load` and :inst:`store` instructions for
 accessing memory, as well as :ref:`extending loads and truncating stores
-<extload-truncstore>`. There are also more restricted operations for accessing
-specific types of memory objects.
+<extload-truncstore>`.
+
+When the address is the result of a :inst:`heap_addr`, :inst:`stack_addr`, or
+:inst:`global_addr` with sufficient size to cover the access given its size
+and offset, and there are no intervening memory reallocations, they are *safe*,
+and either succeed or reliably *trap*, depending on whether the computed address
+is *mapped* or not.
```

Comment:
Instead of using italics, we should define these terms in the glossary and refer to them with `:term:`

Reply:
Ok, I've drafted up possible definitions for "mapped", "trap", and "safe". They're still a little vague in some respects, but I think this takes us in the right direction.